activerecord/lib/active_record/autosave_association.rb

```diff
@@ -176,7 +176,8 @@ def define_non_cyclic_method(name, &block)
         # check if the save or validation methods have already been defined
         # before actually defining them.
         def add_autosave_association_callbacks(reflection)
-          save_method = :"autosave_associated_records_for_#{reflection.name}"
+          method_suffix = reflection.options[:class_name].try(:underscore)
```

Comment:
@egilburg Thank you for spending time to review my pr. You are right, if user does not explicitly set class_name then ```options[:class_name]``` returns nil. But this method name is not visible from outside of the rails, it's an internal method so it does not matter it contains underscore at the end of the it's name or not. Any operation to avoid unnecessary underscore from at the end of the method causes more complexity.